# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../firebird/icu.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./src/common/classes/alloc.h.old	2009-07-09 13:04:14.000000000 +0200
+++ ./src/common/classes/alloc.h	2009-07-09 13:05:37.000000000 +0200
@@ -48,6 +48,8 @@
 					   stdlib.h (EKU) */
 #endif
 
+#include <new>
+
 #ifdef _MSC_VER
 #define THROW_BAD_ALLOC
 #else
@@ -448,15 +450,6 @@
 	);
 }
 
-inline void* operator new(size_t, void* ptr) throw() 
-{
-	return ptr;
-}
-inline void* operator new[](size_t, void* ptr) throw() 
-{
-	return ptr;
-}
-
 inline void operator delete(void* mem) throw()
 {
 	Firebird::MemoryPool::globalFree(mem);

